django_eventstream/views.py

```diff
@@ -32,39 +34,73 @@ def wake_threadsafe(self):
         self.loop.call_soon_threadsafe(self.aevent.set)
 
 
+
+class RedisListener(object):
+    def __init__(self):
+        try:
+            from redis.asyncio import Redis
+        except ImportError:
+            raise ImportError("You must install the redis package to use RedisListener for multiprocess event handling. \n pip install redis")
+
+        self.redis_client = Redis(**settings.EVENTSTREAM_REDIS)
+        self.pubsub = self.redis_client.pubsub()
+
+    async def listen(self):
+        await self.pubsub.subscribe('events_channel')
+        async for message in self.pubsub.listen():
+            if message['type'] == 'message':
+                event_data = json.loads(message['data'])
+                channel = event_data['channel']
+                event_type = event_data['event_type']
+                data = event_data['data']
+                skip_user_ids = event_data['skip_user_ids']
+
+                from .event import Event
+                from .views import get_listener_manager
+
+                e = Event(channel, event_type, data)
+
+                # Notify local listeners
+                get_listener_manager().add_to_queues(channel, e)
+
+    async def start(self):
+        await self.listen()
+
 class ListenerManager(object):
     def __init__(self):
         self.lock = threading.Lock()
         self.listeners_by_channel = {}
+        self.redis_listener = None
+        if hasattr(settings, 'EVENTSTREAM_REDIS'):
+            self.redis_listener = RedisListener()
+
+    async def start_redis_listener(self):
+        await self.redis_listener.start()
 
     def add_listener(self, listener):
-        self.lock.acquire()
-        logger.info(f"added listener {id(listener)}")
-        try:
+        if self.redis_listener:
+            loop = asyncio.get_event_loop()
+            if not self.redis_listener.pubsub.subscribed:
```

Comment:
I think there's a race here. This section is not locked, and even if it was, there is not a guarantee that `self.redis_listener.pubsub.subscribed` evaluates to true immediately after `loop.create_task()` returns. I suggest protecting with a lock and a bool.